python/shared.py

```diff
@@ -0,0 +1,129 @@
+import json
+from dataclasses import dataclass, field
+from typing import Optional, Iterable, Any
+
+from dataclasses_json import config
+
+TASK_QUEUE = "CustomerLoyaltyTaskQueue"
+CUSTOMER_WORKFLOW_ID_FORMAT = "customer-{}"
+EVENT_HISTORY_THRESHOLD = 10000
+
+# Signal and query names
+SIGNAL_CANCEL_ACCOUNT = "cancelAccount"
+SIGNAL_ADD_POINTS = "addLoyaltyPoints"
+SIGNAL_INVITE_GUEST = "inviteGuest"
+SIGNAL_ENSURE_MINIMUM_STATUS = "ensureMinimumStatus"
+QUERY_GET_STATUS = "getStatus"
+QUERY_GET_GUESTS = "getGuests"
+
+
+@dataclass
+class Customer:
+    customerId: str = field(metadata=config(field_name="customerId"))
+    name: Optional[str] = ""
+    points: Optional[int] = 0
+    guests: Optional[set[str]] = field(default_factory=set)
+    account_active: Optional[bool] = field(default_factory=lambda: True, metadata=config(field_name="accountActive"))
+    status: Optional['Status'] = field(default_factory=lambda: Status(0))
```

Comment:
```suggestion
    status: Status = field(default_factory=Status)
```

Same here, if you have a default, no need for optional. Also default constructor is already `0`. Also, if you're having circular type dependency issues, add `from __future__ import annotations` as the first line of the file